decidim-core/app/models/decidim/scope.rb

```diff
@@ -60,19 +60,19 @@ def forbid_cycles
     end
 
     def create_part_of
-      self[:part_of] = calculate_part_of
+      recalculate_part_of
       save if changed?
     end
 
     def update_part_of
-      self[:part_of] = calculate_part_of
+      recalculate_part_of
     end
 
-    def calculate_part_of
+    def recalculate_part_of
```

Comment:
Actually, I'm not sure my suggestion is clear either, since "updating" is usually associated to "saving to DB", but it's not the case here... Naming is hard. :grimacing:

Just leave it as it is unless you have a better suggestion.